core/templates/services/assets-backend-api.service.ts

```diff
@@ -155,7 +155,10 @@ export class AssetsBackendApiService {
   }
 
   getThumbnailUrlForPreview(
-      entityType: string, entityId: string, filename: string): string {
+      entityType: string, entityId: string, filename: string): string | null {
+    if (filename === '') {
+      return null;
+    }
```

Comment:
Can you please explain the reason behind this change? (asking this I was not expecting such change from a migration PR!)

Reply:
Hi @DubeySandeep this function `getThumbnailUrlForPreview` is dependent on UrlInterpolationService. We can't pass null values for filename into our `interpolateUrl` function because it gives `Every parameter passed into interpolateUrl must have string values,but received: {filename: null}`. We are using `getThumbnailUrlForPreview` in our story-viewer-page .If a user creates a story with no thumbnail hence no filename then it throws the above warning.So I had to first check for filename to be null in my component file.If they are null then use `' '` in place of them.So these changes just preserve the logic and also lets thumbnail with empty or null values to also load

Reply:
Hi @DubeySandeep this function `getThumbnailUrlForPreview` is dependent on 'UrlInterpolationService'. We can't pass null values for filename into our interpolateUrl function because it gives `Every parameter passed into interpolateUrl must have string values,but received: {filename: null}`. We are using getThumbnailUrlForPreview in our story-viewer-page .If a user creates a story with no thumbnail hence no filename then it throws the above warning.So I had to first check for filename to be null in my component file.If they are null then use ' ' in place of them.
Without these changes in `AssetsBackendApiService` it results in the following error
![image](https://user-images.githubusercontent.com/60519303/116156685-6d95a500-a709-11eb-9a4e-0854fb7d9a8d.png)
because we have just passed filename to be `' '` in our getThumbnailPreview.
So we can't

1. pass null to our filenames
2. pass `' '` to our filenames

And with changes it looks like:
![image](https://user-images.githubusercontent.com/60519303/116157244-44294900-a70a-11eb-8d17-2a57e4c64104.png)
Just like before
So these changes just preserve the logic and also lets thumbnail with empty or null values to also load

Reply:
@DubeySandeep Reverted this checking for it in the component itself. PTAL!